### PR TITLE
[eclipse/xtext#1569] Refactor Xtend to Java

### PR DESCRIPTION
**Pull request by kthoms**
The usage of Xtend in the Xtext code base should be reduced where the
usage of Xtend is not super beneficial. Replaced Xtend classes by the
compiled code and refactored the code to more readable Java.

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext/src/org/eclipse/xtext/tasks/DefaultTaskTagProvider.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.tasks;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.xbase.lib.CollectionLiterals;
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/tasks/DefaultTaskTagProvider.java]: Unused imports?

**Comment by kthoms** [comment]
Performed more cleanups. @szarnekow Could you have a look?

File: org.eclipse.xtext/src/org/eclipse/xtext/tasks/Task.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.tasks;
+
+/**
+ * @author Stefan Oehme - Initial contribution and API
+ * @since 2.6
+ */
+public class Task {
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/tasks/Task.java]: Not related to this PR, but a `toString` would be nice.

File: org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTags.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	public Pattern toPattern() {
+		if (pattern == null) {
+			int flags = Pattern.MULTILINE;
+			if (!caseSensitive) {
+				flags = flags | Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE;
+			}
+			StringBuilder builder = new StringBuilder();
+			builder.append("^.*((");
+			String tagNames = taskTags.stream().map(it -> Pattern.quote(it.getName())).collect(joining("|"));
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTags.java]: could use `collect(joining(delimiter, prefix, suffix))` and simplify the code.

**Comment by kthoms** [comment]
I'll address the two issues and merge then.

File: org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTag.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.tasks;
+
+/**
+ * @author Stefan Oehme - Initial contribution and API
+ * @since 2.6
+ */
+public class TaskTag {
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTag.java]: Eager merging ... apparently I assumed that you'd add `toString()` here, too

File: org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTags.java (hunk 3 above)
Comment (szarnekow) [reply org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTags.java]: This is still unresolved, isn't it?

Comment (kthoms) [reply org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTags.java]: Actually I wasn't sure what you meant here. collect(joining()) is already used. How would you simplify it?

Comment (szarnekow) [reply org.eclipse.xtext/src/org/eclipse/xtext/tasks/TaskTags.java]: The code builds a new StringBuilder, appends a prefix, uses joining and appends a suffix. joining is overloaded and accepts a prefix and suffix along with the delimiter, too.
You can create the entire string in one shot.

**Pull request merged**